### PR TITLE
feat: add more acs

### PR DESCRIPTION
**Pull request by Jiajia-Cui**

### CODE REVIEW

File: protocol/0009-MRKP-mark_price.md [hunk 1]
```diff
@@ -27,18 +27,17 @@ The mark price isn't updated because `network.markPriceUpdateMaximumFrequency =
...
 Here, we only write the acceptance criteria.
 Note that for calculating the median with an even number of entries we sort, pick out the two values that are in the middle of the list and average those. So in particular with two values a median is the same as the average for our purposes.
 
-
 ## Acceptance criteria
 
 - The mark price must be set when the market leaves opening auction. (<a name="0009-MRKP-002" href="#0009-MRKP-002">0009-MRKP-002</a>)
 - Each time the mark price changes the market data event containing the new mark price should be emitted.Specifically, the mark price set after leaving each auction, every interim mark price as well as the mark price based on last trade used at market termination and the one based on oracle data used for final settlement should all be observable from market data events. (<a name="0009-MRKP-009" href="#0009-MRKP-009">0009-MRKP-009</a>)
+- When leaving the opening auction, if the mark price is undefined then use the opening auction uncrossing price.(<a name="0009-MRKP-001" href="#0009-MRKP-001">0009-MRKP-001</a>)
```
Comment (davidsiska-vega) [inline protocol/0009-MRKP-mark_price.md]:
```suggestion
- If a market mark price is configured in such a way that the mark price methodology hasn't provided a price at the 
point of uncrossing the opening auction, then the auction uncrossing price is set as the first mark price, regardless of what the mark price methodology says. (<a name="0009-MRKP-001" href="#0009-MRKP-001">0009-MRKP-001</a>)
```

File: protocol/0009-MRKP-mark_price.md [hunk 2]
```diff
@@ -70,6 +91,20 @@ Note that for calculating the median with an even number of entries we sort, pic
...
 
+- Set price type to "median", only have data souce availabel from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, check the mark price is correctly calculated (<a name="0009-MRKP-032" href="#0009-MRKP-032">0009-MRKP-032</a>) and a perps market (<a name="0009-MRKP-033" href="#0009-MRKP-033">0009-MRKP-033</a>)
+
+- Set price type to "median", only have data souce availabel from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, and 1 trade at 15940, move time, and check stale price should not be included (<a name="0009-MRKP-032" href="#0009-MRKP-032">0009-MRKP-032</a>) and a perps market (<a name="0009-MRKP-033" href="#0009-MRKP-033">0009-MRKP-033</a>) 
+
+### Validation
+
+- Boundary values are respected for the market parameters
+
+  - `markPriceConfiguration: decayWeight` valid values: `>=0`(<a name="0009-MRKP-050" href="#0009-MRKP-050">0009-MRKP-050</a>)
```
Comment (ze97286) [inline protocol/0009-MRKP-mark_price.md]: I think it must also be <=1

File: protocol/0009-MRKP-mark_price.md [hunk 3]
```diff
@@ -70,6 +91,20 @@ Note that for calculating the median with an even number of entries we sort, pic
...
+
+  - `markPriceConfiguration: decayPower` valid values: `natural numbers but not 0`(<a name="0009-MRKP-051" href="#0009-MRKP-051">0009-MRKP-051</a>)
```
Comment (ze97286) [inline protocol/0009-MRKP-mark_price.md]: I think we said we only allow it to be 0,1,2,3?

**Review by davidsiska-vega — DISMISSED** [review]

File: protocol/0009-MRKP-mark_price.md [hunk 4]
```diff
@@ -69,6 +90,24 @@ Note that for calculating the median with an even number of entries we sort, pic
...
+
+  - `markPriceConfiguration: decayWeight` valid values: `[0,1]`(<a name="0009-MRKP-050" href="#0009-MRKP-050">0009-MRKP-050</a>)
+
+  - `markPriceConfiguration: decayPower` valid values: `{1,2,3}`(<a name="0009-MRKP-051" href="#0009-MRKP-051">0009-MRKP-051</a>)
+
+  - `markPriceConfiguration: cashAmount` valid values: `>=0`(<a name="0009-MRKP-052" href="#0009-MRKP-052">0009-MRKP-052</a>)
+
+  - `markPriceConfiguration: source weight` valid values: `[0,1]`(<a name="0009-MRKP-053" href="#0009-MRKP-053">0009-MRKP-053</a>)
+
+  - `markPriceConfiguration: source staleness tolerance` valid values: `time format: hms`(<a name="0009-MRKP-054" href="#0009-MRKP-054">0009-MRKP-054</a>)
```
Comment (davidsiska-vega) [inline protocol/0009-MRKP-mark_price.md]: This is fine, just checking: Is this how it's implemented? @ze97286

Comment (davidsiska-vega) [reply protocol/0009-MRKP-mark_price.md]: Oh and maybe we should say that it can't be -ve time!

Comment (ze97286) [reply protocol/0009-MRKP-mark_price.md]: it's defined as duration, so must be a valid duration string, e.g. "5s", "24h", etc.

**Review by davidsiska-vega — APPROVED** [review]

File: protocol/0009-MRKP-mark_price.md [hunk 5]
```diff
@@ -69,6 +90,24 @@ Note that for calculating the median with an even number of entries we sort, pic
...
+
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, and 1 trade at 15940, move time, and check stale price should not be included (<a name="0009-MRKP-034" href="#0009-MRKP-034">0009-MRKP-034</a>) and a perps market (<a name="0009-MRKP-035" href="#0009-MRKP-035">0009-MRKP-035</a>)
+
+### Validation
+
+- Boundary values are respected for the market parameters
+
+  - `markPriceConfiguration: decayWeight` valid values: `[0,1]`(<a name="0009-MRKP-050" href="#0009-MRKP-050">0009-MRKP-050</a>)
+
+  - `markPriceConfiguration: decayPower` valid values: `{1,2,3}`(<a name="0009-MRKP-051" href="#0009-MRKP-051">0009-MRKP-051</a>)
```
Comment (ze97286) [inline protocol/0009-MRKP-mark_price.md:103]: Are we removing 0? we don't really need to now

File: protocol/0009-MRKP-mark_price.md [hunk 6]
```diff
@@ -69,6 +90,24 @@ Note that for calculating the median with an even number of entries we sort, pic
...
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, check the mark price is correctly calculated (<a name="0009-MRKP-032" href="#0009-MRKP-032">0009-MRKP-032</a>) and a perps market (<a name="0009-MRKP-033" href="#0009-MRKP-033">0009-MRKP-033</a>)
+
+- Set price type to "median", only have data source available from "Trade-size-weighted average price" and "Leverage-notional book price" and 1 trade at 15920, and 1 trade at 15940, move time, and check stale price should not be included (<a name="0009-MRKP-034" href="#0009-MRKP-034">0009-MRKP-034</a>) and a perps market (<a name="0009-MRKP-035" href="#0009-MRKP-035">0009-MRKP-035</a>)
+
+### Validation
```
Comment (ze97286) [inline protocol/0009-MRKP-mark_price.md:97]: another family of validations is that the length of the arrays must agree. e.g. if more than one oracle is defined then the length of the weights and staleness array must be inline with that.

Comment (davidsiska-vega) [reply protocol/0009-MRKP-mark_price.md:97]: I only care that I can get no time decay i.e. K(s,t) = 1, which I can do with \alpha= 0 and p = 1 (or any other value).

File: protocol/0009-MRKP-mark_price.md (hunk 5 above)
Comment (davidsiska-vega) [reply protocol/0009-MRKP-mark_price.md:103]: I only care that I can get no time decay i.e. K(s,t) = 1, which I can do with \alpha= 0 and p = 1 (or any other value).

Comment (ze97286) [reply protocol/0009-MRKP-mark_price.md:103]: I don't understand. there's nothing wrong with K=1, I'd think the problem is with k=0 for any trade time s which you get from power=0 and alpha=1

File: protocol/0009-MRKP-mark_price.md (hunk 6 above)
Comment (ze97286) [reply protocol/0009-MRKP-mark_price.md:97]: something for fuzzing I guess.

Comment (davidsiska-vega) [reply protocol/0009-MRKP-mark_price.md:97]: In separate PR.

**Pull request merged**